src/Dropdown/Dropdown.tsx

```diff
@@ -74,9 +72,6 @@ export const Dropdown = forwardRef(function Dropdown(
       onSelect(list[0].value)
     }
 
-    // update 'key' props in order to deselect the active option if a new list passed
-    list.length > 1 ? setKey(`${list[0].value}-${list.length}`) : ''
```

Comment:
This was a fix from 3 years ago but couldn't reproduce
This seems more like an hack than a real fix so I was thinking of removing it for now